### PR TITLE
refactor: remove interfaces generating entity REST resources.

### LINKED ISSUE #94: Replace PanacheEntityResource with PanacheEntityBase
## Enhancement

As most of the REST crud primaza operations will return HTML Text, will consume HTML Forms, etc then it is not needed to generate the default CRUD REST JSON endpoints using PanacheEntityResource as JSON is only used for testing purpose.

So I suggest that we remove the interface extending PanacheEntityResource and that we use for each entity class `PanacheEntityBase`

WDYT ? @Sgitario @aureamunoz

### PR DESCRIPTION
**Pull request by aureamunoz**
Fixes #94

### CODE REVIEW

File: servicebox-app/src/main/java/io/halkyon/resource/page/ClaimResource.java [hunk 1]
```diff
@@ -68,12 +68,17 @@ private TemplateInstance showList(List<Claim> claims) {
     @Consumes(MediaType.APPLICATION_JSON)
     @Path("/name/{name}")
     public io.halkyon.model.Claim findByName(@PathParam("name") String name) {
-        return io.halkyon.model.Claim.findByName(name);
+        Claim claim = Claim.findByName(name);
+        if (claim == null) {
+            throw new WebApplicationException("Claim with name " + name + " does not exist.", 404);
```
Comment (Sgitario) [inline servicebox-app/src/main/java/io/halkyon/resource/page/ClaimResource.java:73]: we should throw the NotFoundException exception instead.

File: servicebox-app/src/main/java/io/halkyon/resource/page/ServiceResource.java [hunk 2]
```diff
@@ -83,7 +84,11 @@ private TemplateInstance showList(List<Service> services) {
     @Consumes(MediaType.APPLICATION_JSON)
     @Path("/name/{name}")
     public io.halkyon.model.Service findByName(@PathParam("name") String name) {
-        return io.halkyon.model.Service.findByName(name);
+        Service service = Service.findByName(name);
+        if (service == null) {
+            throw new WebApplicationException("Service with name " + name + " does not exist.", 404);
```
Comment (Sgitario) [inline servicebox-app/src/main/java/io/halkyon/resource/page/ServiceResource.java:89]: we should throw the NotFoundException exception instead.

File: servicebox-app/src/test/java/io/halkyon/ClaimsEndpointTest.java [hunk 3]
```diff
@@ -63,48 +44,37 @@ public void testQueryUsingNameToGetClaims(){
...
+    public void claimCreatedViaForm(){
         // An htmx request will contain a HX-Request header and Content-Type: application/x-www-form-urlencoded
+        String claimName = "mysql-claim";
         given()
                 .header("HX-Request", true)
                 .contentType(MediaType.APPLICATION_FORM_URLENCODED)
-                .body("{\"name\": \"mysql-demo\", \"serviceRequested\": \"mysql:3306\"}")
+                .formParam("name", claimName)
+                .formParam("serviceRequested","mysql-8.0.3")
+                .formParam("status","new")
```
Comment (Sgitario) [inline servicebox-app/src/test/java/io/halkyon/ClaimsEndpointTest.java:61]: we don't need to provide the `status` field here.

File: servicebox-app/src/test/java/io/halkyon/ServicesEndpointTest.java [hunk 4]
```diff
@@ -1,91 +1,81 @@
...
-                .statusCode(201);
-
-    }
-
-    @Test
-    public void testAddServiceViaHtmxForm(){
+    public void serviceIsFoundByName() {
         // An htmx request will contain a HX-Request header and Content-Type: application/x-www-form-urlencoded
+
+        String rabbitMQ4 = "RabbitMQ2";
```
Comment (Sgitario) [inline servicebox-app/src/test/java/io/halkyon/ServicesEndpointTest.java:25]: it should be "RabbitMQ4"?

File: servicebox-app/src/test/java/io/halkyon/ServicesEndpointTest.java [hunk 5]
```diff
@@ -1,91 +1,81 @@
...
         given()
                 .header("HX-Request", true)
                 .contentType(MediaType.APPLICATION_FORM_URLENCODED)
-                .body("{\"name\": \"RabbitMQ2\", \"version\": \"3.11.2\", \"endpoint\": \"tcp:5672\", \"deployed\": \"false\" }")
+                .formParam("name", rabbitMQ4)
+                .formParam("version", "3.11.1")
+                .formParam("endpoint", "tcp:5672")
+                .formParam("deployed", false)
```
Comment (Sgitario) [inline servicebox-app/src/test/java/io/halkyon/ServicesEndpointTest.java:32]: we should not provide the `deployed` field.

File: servicebox-app/src/test/java/io/halkyon/ServicesEndpointTest.java [hunk 6]
```diff
@@ -1,91 +1,81 @@
...
         given()
-                .contentType(MediaType.APPLICATION_JSON)
-                .accept("application/json")
-                .body(request)
+                .header("HX-Request", true)
+                .contentType(MediaType.APPLICATION_FORM_URLENCODED)
+                .formParam("name", rabbitMQ4)
+                .formParam("version", "3.11.1")
+                .formParam("endpoint", "tcp:5672")
+                .formParam("deployed", false)
```
Comment (Sgitario) [inline servicebox-app/src/test/java/io/halkyon/ServicesEndpointTest.java:57]: we should not provide the `deployed` field.

File: servicebox-app/src/test/java/io/halkyon/utils/TestUtils.java [hunk 7]
```diff
@@ -30,31 +37,53 @@ public static Service createService(String serviceName, String serviceVersion, b
     }
 
     public static Service createService(String serviceName, String serviceVersion, String endpoint) {
-        return createService(serviceName, serviceVersion, endpoint, false);
+         return createService(serviceName, serviceVersion, endpoint, false);
```
Comment (Sgitario) [inline servicebox-app/src/test/java/io/halkyon/utils/TestUtils.java:40]: I think this is adding an extra space at the beginning.

**Review by Sgitario — CHANGES_REQUESTED** [review]

**Comment by aureamunoz** [comment]
Thanks for the review, I took your comments into account and rebase onto main branch. It should be good, let's see what CI says

**Comment by Sgitario** [comment]
These comments are still unresolved: https://github.com/halkyonio/primaza-poc/pull/142#discussion_r1028948284

**Comment by Sgitario** [comment]
These changes are breaking the CI: https://github.com/halkyonio/primaza-poc/actions/runs/3521610195/jobs/5903673217#step:8:70
I will update the CI to spot it property as a failure.

File: servicebox-app/src/main/java/io/halkyon/resource/page/ClaimResource.java (hunk 1 above)
Comment (cmoulliard) [reply servicebox-app/src/main/java/io/halkyon/resource/page/ClaimResource.java]: Take care that throwing an exception will return a HTTP 404 exception that it must be handled by HTMX. Ideally if there is such an error, we should like when we submit an error which is raising an error that a message is displayed in a box in red within the HTML page. See what we did for HTML Claim form by example

**Review by Sgitario — APPROVED** [review]

**Pull request merged**